### ISSUE TITLE
Error handling of base64 decoding when consuming ByteArray 

### ISSUE THREAD

**Issue by martijnvermaat** [arskom/spyne#475]
When Spyne expects base64 encoded data but I feed it something that cannot be decoded as such, the `TypeError` raised by `b64decode` isn't handled.

For example, use the following minimal adaptation of the spyne.io example:

```python
from spyne.application import Application
from spyne.decorator import srpc
from spyne.service import ServiceBase
from spyne.model.primitive import Integer
from spyne.model.binary import ByteArray

from spyne.protocol.http import HttpRpc
from spyne.protocol.json import JsonDocument

from spyne.server.wsgi import WsgiApplication

class HelloWorldService(ServiceBase):
    @srpc(ByteArray, _returns=Integer)
    def say_hello(name):
        return 42

application = Application([HelloWorldService],
    tns='spyne.examples.hello',
    in_protocol=HttpRpc(validator='soft'),
    out_protocol=JsonDocument()
)

if __name__ == '__main__':
    # You can use any Wsgi server. Here, we chose
    # Python's built-in wsgi server but you're not
    # supposed to use it in production.
    from wsgiref.simple_server import make_server

    wsgi_app = WsgiApplication(application)
    server = make_server('0.0.0.0', 8000, wsgi_app)
    server.serve_forever()
```

Now, feed it something that's not in base64:

```
$ curl 'http://localhost:8000/say_hello?name=abcde'
A server error occurred.  Please contact the administrator.
```

```
$ python serve.py 
Traceback (most recent call last):
  File "/usr/lib/python2.7/wsgiref/handlers.py", line 85, in run
    self.result = application(self.environ, self.start_response)
  File "/home/martijn/.virtualenvs/mutalyzer/local/lib/python2.7/site-packages/spyne/server/wsgi.py", line 244, in __call__
    return self.handle_rpc(req_env, start_response)
  File "/home/martijn/.virtualenvs/mutalyzer/local/lib/python2.7/site-packages/spyne/server/wsgi.py", line 356, in handle_rpc
    self.get_in_object(p_ctx)
  File "/home/martijn/.virtualenvs/mutalyzer/local/lib/python2.7/site-packages/spyne/server/_base.py", line 93, in get_in_object
    message=self.app.in_protocol.REQUEST)
  File "/home/martijn/.virtualenvs/mutalyzer/local/lib/python2.7/site-packages/spyne/protocol/http.py", line 199, in deserialize
    ctx.descriptor.in_message, self.validator, req_enc=req_enc)
  File "/home/martijn/.virtualenvs/mutalyzer/local/lib/python2.7/site-packages/spyne/protocol/dictdoc/simple.py", line 198, in simple_dict_to_object
    req_enc, validator)
  File "/home/martijn/.virtualenvs/mutalyzer/local/lib/python2.7/site-packages/spyne/protocol/dictdoc/simple.py", line 128, in _to_native_values
    self.binary_encoding)
  File "/home/martijn/.virtualenvs/mutalyzer/local/lib/python2.7/site-packages/spyne/protocol/_inbase.py", line 245, in from_unicode
    return handler(class_, string, *args, **kwargs)
  File "/home/martijn/.virtualenvs/mutalyzer/local/lib/python2.7/site-packages/spyne/protocol/_inbase.py", line 465, in byte_array_from_string
    return binary_decoding_handlers[encoding](value)
  File "/home/martijn/.virtualenvs/mutalyzer/local/lib/python2.7/site-packages/spyne/model/binary.py", line 129, in from_urlsafe_base64
    return [urlsafe_b64decode(_bytes_join(value))]
  File "/usr/lib/python2.7/base64.py", line 112, in urlsafe_b64decode
    return b64decode(s, '-_')
  File "/usr/lib/python2.7/base64.py", line 76, in b64decode
    raise TypeError(msg)
TypeError: Incorrect padding
127.0.0.1 - - [16/Dec/2015 12:10:38] "GET /say_hello?name=abcde HTTP/1.1" 500 59
```

I don't think this is an error we can easily handle outside of Spyne.

**Comment by plq** [comment]
Hello, thanks for the report.

What do you think about this change?

**Comment by martijnvermaat** [comment]
LGTM :+1:

**Comment by plq** [comment]
Patch to be merged in #476. closing this one.
